### PR TITLE
[Feature PR]: Add `noThrow` option

### LINKED ISSUE #169: [Feature Request]: Allow return "errors" instead of throwing
Hello and thanks for this library.
For my case, this library improves Request/second by 3-7% which is good for me.
But when schema fails, i want get error by string and show to users.

Using try/catch decreases performance by ~40% for my app. Is there any way get it or available options?

PS: I'm not using `fastify`, i made own library :)

### PR DESCRIPTION
**Pull request by dalisoft**
This PR closes #169 

Related issue: #169

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,7 +28,9 @@ function isValidSchema (schema, name) {
       name = ''
     }
     const first = validate.errors[0]
-    const err = new Error(`${name}schema is invalid: data${first.dataPath} ${first.message}`)
+    const err = new Error(
+      `${name}schema is invalid: data${first.dataPath} ${first.message}`
+    )
```
Comment (mcollina) [inline index.js]: Can you avoid changing the format of the code?

Comment (dalisoft) [reply index.js]: I will try, but this formatted with `standard` code-style. I manually didn't touched this part

Comment (dalisoft) [reply index.js]: I'm formatted with `npx standard --fix`

**Comment by dalisoft** [comment]
For objects and string should work, for string and numbers not required like objects, because strings or numbers can be manually filtered.
Using `try/catch` reduces performance by ~50% (after optimising other parts of code) which is not good.

For `fastify` same performance degaration should be apply as `fastify` gets error? (I not sure about this, but i think yes)

**Comment by mcollina** [comment]
I don't understand the point you are making.

I agree that supporting this pattern is a good thing, but we should returning the data through the same value.  As an example check: https://www.npmjs.com/package/is-my-json-valid#verbose-mode-shows-more-information-about-the-source-of-the-error.

**Comment by dalisoft** [comment]
Yep, thats what i want. Without errors (`throw`), only returning `.errors`

**Comment by jsumners** [comment]
The only viable way I see is as outlined in https://medium.com/knock-engineering/happy-path-error-handling-1f698068a345

**Comment by dalisoft** [comment]
@jsumners Thanks for help, but i'm trying improve my library RPS by count (which is matters), i like `fast-json-stringify` and works very very best, but when data mismatches, performance is reducing by `fast-json-stringify` `throw new Error(...)` which i'm don't want

**Comment by jsumners** [comment]
My suggestion has nothing to do with throwing an error. I am suggesting a complete change in the returned data type:

```js
const result = fastStringify({foo: 'bar'})
if (result.error) {
  console.error(result.error.message)
  process.exit(1)
}
console.log(result.value) // `{"foo":"bar"}`
```

**Comment by dalisoft** [comment]
Wow, sorry, looks I didn’t understand good. Now understand. +1

**Comment by mcollina** [comment]
@dalisoft would you mind to update this PR to match https://github.com/fastify/fast-json-stringify/pull/172#issuecomment-498245616?

**Comment by dalisoft** [comment]
@mcollina I don't think i can, because i don't know how to push `errors` or `return with like this`. Maybe i other OPEN when i can do this? (but not in few weeks)

**Comment by mcollina** [comment]
I'm sorry for the long delay, I didn't see this. Are you still interesting in pushing this forward?

File: index.js [hunk 2]
```diff
@@ -351,9 +338,16 @@ function addPatternProperties (schema, externalSchema, fullSchema) {
           json += $asString(keys[i]) + ':' + $asBoolean(obj[keys[i]])
       `
     } else {
-      code += `
+      if (options && options.noThrow) {
+        code += `
+        ${addComma}
+        return {error: "Cannot coerce " + obj[keys[i]] + " to ${type}"}
```
Comment (jsumners) [inline index.js:344]: Why isn't the value of the `error` property an instance of `Error`?

Comment (dalisoft) [reply index.js:344]: Sometimes i wanted to get only object instead of Error improve performance before

**Comment by dalisoft** [comment]
As i deprecated (maybe currently) project, I don’t think this feature need. 
I used default API and worked fine last time i tried except invalid response mismatching performance drop

File: index.js (hunk 2 above)
Comment (jsumners) [reply index.js:344]: I would be okay with this change if the error property is an instance of the Error object.

**Pull request closed without merging**